### DISCUSSION TITLE
Nokogiri builder - HTML5 self-enclosing tags issue

### DISCUSSION THREAD

**Discussion by sb934** [sparklemotion/nokogiri#2405] [General Q&A] [answered]
Hello, I'm using the HTML builder to write an HTML doc. However, the HTML created is broken with self-closing tags which are not valid in HTML5 anymore. Reviewing the issues discussed, I believe this has been fixed however, I'm still struggling to point where I'm going wrong. If anyone has faced a similar issue and can guide me with it. 
The ruby code also works fine on some machines with correct tags created but breaks HTML on some. Let me know whatever supporting info is required. I'm using `Nokogiri:HTML::Document.new` to create an HTML doc and `Nokogiri::HTML::Builder` for writing it. I end up with a DOCTYPE of HTML 4.0 with this.

**Comment by flavorjones** [comment]
@sb934 Can you please open a new "bug report" or "help request" issue, and fill out the template with the requested information?

**Reply by sb934** [reply, marked as answer]
Aah never mind, I figured it. Using to_xhtml() rather than to_html() on the doc was causing it. Thank you for your response though.

**Comment by BeAhMa** [comment]
// src/mocks.js
import { setupWorker, rest } from 'msw'

const worker = setupWorker(
  rest.post('/login', (req, res, ctx) => {
    const isAuthenticated = sessionStorage.getItem('username')

    if (!isAuthenticated) {
      return res(
        ctx.status(403),
        ctx.json({
          errorMessage: 'Not authenticated',
        }),
      )
    }

    return res(
      ctx.json({
        firstName: 'John',
      }),
    )
  }),
)

// Register the Service Worker and enable the mocking
worker.start()

**Reply by BeAhMa** [reply]
[]()
